editors/ccap-temple-mode.el

```diff
@@ -1,3 +1,148 @@
+;; --------------------------------------------------------------------
+;; Customization
+
+(defcustom ccap-temple-mode-codegen-repo ""
+  "Path to the codegen repository directory."
+  :group 'ccap-temple-mode
+  :type 'string)
+
+(defconst ccap-temple-out-buffer (generate-new-buffer "*ccap-temple output*"))
+
+;; --------------------------------------------------------------------
+;; Helpers
+
+(defun ccap-temple-current-line ()
+  "Get the current line as text."
+  (buffer-substring-no-properties
+   (line-beginning-position)
+   (line-end-position)))
+
+(defun ccap-temple-get-main-dir (tmpl-filename)
+  "Given a TMPL-FILENAME, get the directory for the main src folder.
+Assumes that the tmpl file is in <project-dir>/main/resources/."
+  (car (split-string tmpl-filename "/resources")))
```

Comment:
That should not be the right path... But  I think referring to the IDL files as templates has led to confusion...

Reply:
Yeah... I'm just calling them 'tmpl' files because that's the extension. FWIW this does generate the correct path, at least for how it's structured in scca-dash.

Reply:
Yeah... The point is that it's structured wrong in scca-dash. These are not resources that we need at runtime.